boot/zephyr/boards/thingy53_nrf5340_cpuapp.conf

```diff
@@ -64,3 +66,11 @@ CONFIG_RESET_ON_FATAL_ERROR=n
 CONFIG_SPI=n
 CONFIG_I2C=n
 CONFIG_UART_NRFX=n
+
+# The following configurations are required to support simultaneous multi image update
+CONFIG_PCD_APP=y
+CONFIG_UPDATEABLE_IMAGE_NUMBER=2
+CONFIG_BOOT_UPGRADE_ONLY=y
+CONFIG_FLASH_SIMULATOR=y
+CONFIG_FLASH_SIMULATOR_DOUBLE_WRITES=y
+CONFIG_FLASH_SIMULATOR_STATS=n
```

Comment:
Why is flash simulator needed for this ?
https://github.com/nrfconnect/sdk-zephyr/blob/14f880944a81c1ac4796dbf21c325c26b2d06ceb/drivers/flash/Kconfig.simulator#L6

No info is provided in this file and the commit message doesn't describe it either.
Please explain why this is needed.

Reply:
> 
> No info is provided in this file and the commit message doesn't describe it either.
> Please explain why this is needed.

The upload to nrf53 netcore goes through a "window" that the flash simulator provides. There is no direct access to the flash for the netcore update.

Reply:
Then I request three things regarding this:
1. Provide detailed information in the commit message so that someone wondering why this is added can understand when using `git log`.
2. A comment directly in the prj.conf.
3. I think using `FLASH_SIMULATOR` for such purpose is principally wrong, but accepted for this particular PR if you create a followup task. Even if we are providing / re-using the simulator implementation, I think we should provide dedicated nRF Kconfig settings for this use-case.
    That will make it clearer for our useers, as well as we can provide required selects (maybe even with prompt-less symbols) for a better user-experience.

Reply:
For example here: https://github.com/nrfconnect/sdk-zephyr/blob/14f880944a81c1ac4796dbf21c325c26b2d06ceb/drivers/flash/Kconfig.nrf

Reply:
Added explanation in comments and created the requested Jira ticket.

Reply:
@Rallare could you please take a look here.

Reply:
I agree with the fundamental thought here, which is that the appcore needs a "ramdisk" storage for the net-core's binary. This is how the hardware expects sharing of data, so we need to work with it. If you add this to the overall description, it is much easier to understand why it's done.

Is the configurations exposed to users, both cmake and kconfig, in this PR expected to change in the future? 
If yes; add a temp-hack comment on it. It is important that we do not merge anything that we already know will be re-named or even re-written in the near future.

Reply:
I fully understood that the configuration is actually needed for nRF5340 hardware in general.

Reply:
> It is important that we do not merge features that we already know will be re-named or even re-written in the near future without explicitly stating so.

I completely agree on this, it's a hidden temphack :disappointed: . Unfortunately this has already been merged upstream: https://github.com/mcu-tools/mcuboot/pull/1136
meaning the problem exists upstream (and must be cleaned up upstream).
@MarekPieta when a commit is merged upstream, the commit title should be changed to `[nrf fromtree]`.

Reply:
The configuration may be improved in the future, but I would prefer to have a common MCUboot configuration that would be used for Thingy:53 apps now (so we must use existing configs).
For example the flash simulator name may actually change in the future (it was already requested)

Reply:
@tejlmand, using fromtree now. The PRs were merged in meantime.